ops/testing.py

```diff
@@ -1926,30 +1928,26 @@ def __init__(self, framework: ops.Framework):
                 def _on_start(self, event: ops.StartEvent):
                     self.cloud_spec = self.model.get_cloud_spec()
 
-            class TestCharm(unittest.TestCase):
-                def setUp(self):
-                    self.harness = ops.testing.Harness(MyVMCharm)
-                    self.addCleanup(self.harness.cleanup)
-
-                def test_start(self):
-                    cloud_spec_dict = {
-                        'name': 'localhost',
-                        'type': 'lxd',
-                        'endpoint': 'https://127.0.0.1:8443',
-                        'credential': {
-                            'authtype': 'certificate',
-                            'attrs': {
-                                'client-cert': 'foo',
-                                'client-key': 'bar',
-                                'server-cert': 'baz'
-                            },
+            # test_charm.py
+            def test_start(harness):
+                cloud_spec_dict = {
```

Comment:
I would have this like:

```suggestion
                cloud_spec = ops.CloudSpec.from_dict({
```

with the closing brace at the end, and then you can get rid of `expected` and just pass `cloud_spec` to `set_cloud_spec()` and use it in the `assert` statement.

Reply:
Yes it's a good idea!